spec/support/system_helper.rb

```diff
@@ -0,0 +1,17 @@
+module SystemHelper
+  def login
+    user = create(:user)
+
+    visit login_path
+    fill_in 'メールアドレス', with: user.email
+    fill_in 'パスワード', with: '12345678'
+    click_button 'ログイン'
+  end
+  
+  def login_as(user)
+    visit login_path
+    fill_in 'メールアドレス', with: user.email
+    fill_in 'パスワード', with: '12345678'
+    click_button 'ログイン'
+  end
```

Comment:
moduleで共通化するのはいいですね！
moduleで共通化するという意味では、Railsのconcernでも同様のことをやるので、知っておくとよいと思います！
[concernsディレクトリについて\[Rails\] \- haayaaa’s diary](https://haayaaa.hatenablog.com/entry/2018/12/12/140856)

Reply:
あと、ここはこうした方がすっきりしそうです！
```suggestion
  def login(user = nil)
    user = create(:user) if user.nil?

    visit login_path
    fill_in 'メールアドレス', with: user.email
    fill_in 'パスワード', with: '12345678'
    click_button 'ログイン'
  end
```

Reply:
デフォルト引数やキーワード引数でぜひ調べてみてください！

Reply:
ありがとうございます！
ここは解答例を脳死で写経してますね汗
こうしたら更に良くなるんじゃ？と常に考えることが大事だと思いました（戒め）
デフォルト引数とキーワード引数について書いてみました

###### デフォルト引数
```ruby
def login(user = nil)
```
ここの`user = nil`の部分で設定している。
メソッドを呼び出す際に、引数を指定していればそれが渡される
引数を指定しなかった場合はここで設定した値が使用される

###### キーワード引数
メソッドの定義の際に引数を`シンボル:`で指定するとキーワード引数にできる
呼び出す際に引数にキーワードを付けて呼ぶことができるので、引数が何を表すのか、理解しやすくなる
また、引数の順番も入れ替えることができる
```ruby
def profile(name:, age:)
  "私の名前は#{name}です。#{age}歳です。"
end

puts profile(name: 'miyaichi', age: 32)
```


あと、関係ない話なんですが、Github上でコードの校正ってみけたさんがやってる感じでできるの初めて知りました！

Reply:
おお、丁寧に調べていていいですね！
まとめていただいたとおりの認識でよいと思います！
（あと、チェリー本持っているようであれば、該当箇所を一読するとよいと思います）

> あと、関係ない話なんですが、Github上でコードの校正ってみけたさんがやってる感じでできるの初めて知りました！
私も、職場の人が使っているのをきっかけに最近知りました笑